components/concept/awesomebar/src/main/java/mozilla/components/concept/awesomebar/AwesomeBar.kt

```diff
@@ -108,31 +132,29 @@ interface AwesomeBar {
         /**
          * Fired whenever the user changes their input, after they have started interacting with the awesome bar.
          *
-         * This is a synchronous call. It is up to the [AwesomeBar] implementation to perform this operation on a
-         * non-main thread. For example the `browser-awesomebar` component wrap this call in an async block inside a
-         * coroutine:
-         *
-         * ```Kotlin
-         * jobs = providers.map { provider ->
-         *     launch(UI) {
-         *        val suggestions = async(dispatcher) { provider.onInputChanged(text) }
-         *        // Do something with suggestions
-         *     }
-         * }
-         * ```
+         * This is a suspending function. An [AwesomeBar] implementation is expected to invoke this method from a
+         * [Coroutine](https://kotlinlang.org/docs/reference/coroutines-overview.html). This allows the [AwesomeBar]
+         * implementation to group and cancel calls to multiple providers.
          *
-         * This allows the awesome bar component to be in control of the jobs launched and lets it cancel jobs as soon
-         * as they are no longer needed. For the provider this makes the implementation easier as it does not need to
-         * deal with asynchronous cancellable tasks.
+         * Coroutine cancellation is cooperative. A coroutine code has to cooperate to be cancellable:
+         * https://github.com/Kotlin/kotlinx.coroutines/blob/master/docs/cancellation-and-timeouts.md
          *
          * @param text The current user input in the toolbar.
          * @return A list of suggestions to be displayed by the [AwesomeBar].
          */
-        fun onInputChanged(text: String): List<Suggestion>
+        suspend fun onInputChanged(text: String): List<Suggestion>
 
         /**
          * Fired when the user has cancelled their interaction with the awesome bar.
          */
         fun onInputCancelled() = Unit
+
+        /**
+         * If true an [AwesomeBar] implementation can clear the previous suggestions of this provider as soon as the
+         * user continues to type. If this is false an [AwesomeBar] implementation is allowed to keep the previous
+         * suggestions around until the provider returns a new list of suggestions for the updated text.
+         */
+        val shouldClearSuggestions
```

Comment:
Not sure why you need this, but I guess you do!

Reply:
Some suggestions (like the search engine suggestions) should not disappear / re-appear whenever the text changes.